cupyx/scipy/interpolate/_polyint.py

```diff
@@ -0,0 +1,323 @@
+        at any time.
+
+        Parameters
+        ----------
+        yi : cupy.ndarray
+            The y-coordinates of the points the polynomial should pass
+            through. If None, the y values will be supplied later.
+        axis : int, optional
+            Axis in the yi array corresponding to the x-coordinate values
+
+        """
+
+        if yi is None:
+            self.yi = None
+            return
+        self._set_yi(yi, xi=self.xi, axis=axis)
+        self.yi = self._reshape_yi(yi)
+        self.n, self.r = self.yi.shape
+
+    def add_xi(self, xi, yi=None):
+        """Add more x values to the set to be interpolated.
+
+        The barycentric interpolation algorithm allows easy updating
+        by adding more points for the polynomial to pass through.
+
+        Parameters
+        ----------
+        xi : cupy.ndarray
+            The x-coordinates of the points that the polynomial should
+            pass through
+        yi : cupy.ndarray, optional
+            The y-coordinates of the points the polynomial should pass
+            through. Should have shape ``(xi.size, R)``; if R > 1 then
+            the polynomial is vector-valued
+            If `yi` is not given, the y values will be supplied later.
+            `yi` should be given if and only if the interpolator has y
+            values specified
+
+        """
+
+        if yi is not None:
+            if self.yi is None:
+                raise ValueError("No previous yi value to update!")
+            yi = self._reshape_yi(yi, check=True)
+            self.yi = cupy.vstack((self.yi, yi))
+        else:
+            if self.yi is not None:
+                raise ValueError("No update to yi provided!")
+        old_n = self.n
+        self.xi = cupy.concatenate((self.xi, xi))
+        self.n = len(self.xi)
+        self.wi **= -1
+        old_wi = self.wi
+        self.wi = cupy.zeros(self.n)
+        self.wi[:old_n] = old_wi
+        for j in range(old_n, self.n):
+            self.wi[:j] *= self._inv_capacity * (self.xi[j] - self.xi[:j])
+            self.wi[j] = cupy.prod(
+                self._inv_capacity * (self.xi[:j] - self.xi[j])
+            )
+        self.wi **= -1
+
+    def __call__(self, x):
+        """Evaluate the interpolating polynomial at the points x.
+
+        Parameters
+        ----------
+        x : cupy.ndarray
+            Points to evaluate the interpolant at
+
+        Returns
+        -------
+        y : cupy.ndarray
+            Interpolated values. Shape is determined by replacing the
+            interpolation axis in the original array with the shape of x
+
+        Notes
+        -----
+        Currently the code computes an outer product between x and the
+        weights, that is, it constructs an intermediate array of size
+        N by len(x), where N is the degree of the polynomial.
+
+        """
+
+        return _Interpolator1D.__call__(self, x)
```

Comment:
nit: shouldn't we just do this?
```suggestion
        return super().__call__(x)
```

Reply:
Hi, @leofang! Thanks for your input! It works fine. I'll update it.